### PR TITLE
Update Legacy Endpoints for Delete and Edit

### PR DESCRIPTION
**Pull request by MushuEE**
Started work on updating Legacy endpoints for prow Github client. Updated references also. 

There was a unit test missing for Delete so I added that as well.

/area Prow
/kind cleanup

see #19186 for more context. More end point TODO

image

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -2984,7 +2984,7 @@ func (c *client) ListTeamMembers(id int, role string) ([]TeamMember, error) {
 
 // ListTeamRepos gets a list of team repos for the given team id
 //
-// https://developer.github.com/v3/teams/#list-team-repos
+// https://docs.github.com/en/rest/reference/teams#list-team-repos
```
Comment (MushuEE) [inline prow/github/client.go:2987]: Change to `list-team-repositories`

File: prow/github/client.go [hunk 2]
```diff
@@ -3025,7 +3025,7 @@ func (c *client) ListTeamRepos(id int) ([]Repo, error) {
 
 // UpdateTeamRepo adds the repo to the team with the provided role.
 //
-// https://developer.github.com/v3/teams/#add-or-update-team-repository
+// https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository
```
Comment (MushuEE) [inline prow/github/client.go:3028]: `add-or-update-team-repository-permissions`

File: prow/github/client.go [hunk 3]
```diff
@@ -3051,7 +3051,7 @@ func (c *client) UpdateTeamRepo(id int, org, repo string, permission RepoPermiss
 
 // RemoveTeamRepo removes the team from the repo.
 //
-// https://developer.github.com/v3/teams/#add-or-update-team-repository
+// https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository
```
Comment (MushuEE) [inline prow/github/client.go:3054]: Ditto

File: prow/github/client.go [hunk 4]
```diff
@@ -3676,7 +3676,7 @@ func (c *client) TeamHasMember(teamID int, memberLogin string) (bool, error) {
 
 // GetTeamBySlug returns information about that team
 //
-// See https://developer.github.com/v3/teams/#get-team-by-name
+// See https://docs.github.com/en/rest/reference/teams#get-team-by-name
```
Comment (MushuEE) [inline prow/github/client.go:3679]: `get-a-team-by-name`

**Review by fejta — APPROVED** [review]

**Pull request merged**